lib/apiservers/engine/backends/container.go

```diff
@@ -1436,6 +1436,17 @@ func setCreateConfigOptions(config, imageConfig *containertypes.Config) {
 		config.Entrypoint = imageConfig.Entrypoint
 	}
 
+	if config.Volumes == nil {
+		config.Volumes = imageConfig.Volumes
+	} else {
+		for k, v := range imageConfig.Volumes {
+			//NOTE: the value of the map is an empty struct.
+			//      we also do not care about duplicates.
+			//      This Volumes map is really a Set.
+			config.Volumes[k] = v
```

Comment:
Yes that data is stored in the HostConfig.Binds, I believe this is because docker only approaches volumes from a mostly `bind mount` perspective. So the "specified volumes" are processed separately from the anonymous volumes. and then a final set is distilled from that taking the specified volumes over the anonymous ones. The image volumes actually come from the images `config.volumes` set. so those must be joined before the actual fields are processed.